Ashish_P/perview/src/render_components/PerviewsLayouts/NarrowPerview.js

```diff
@@ -70,7 +70,7 @@ const NarrowPerview = ({ currentUserId, perviews, createComment, deleteComment,
             <span>{`${item.data.listPrice.formattedAmount}`}</span>
 
             <a className="buy-btn" href={item.data.detailPageUrl} target="_blank">
-              Buy
+              Buy {!!(item.siteName) ? "on " + item.siteName : ""}
```

Comment:
Hey Ashish, you can use "\`" to concatenate strings with js with ES6 (e.g. \`on${item.siteName}\`).